Removes  django-angular lib. and its usage from login form

DubeySandeep opened the pull request:
### Description
The changes in the PR remove the usage of django-angular (djng) third-party lib. as we use patched lib. and the updated lib. requires lengthy migration and currently, we are using djng only for login form. 

Ticket: https://tasks.opencraft.com/browse/SE-3732


### Testing instruction
- Run the ocim backend server (`vagrant ssh` followed by `make run.dev`)
- Open http://localhost:5000/login 
- FIllup the login form with valid/invalid data for testing
- Open https://stage.manage.opencraft.com/login to compare the UI.

### UI Screenshots
#### Earlier
![image](https://user-images.githubusercontent.com/16653571/157828245-3d4ef4b5-9844-4df6-a0a1-970c230a5c6e.png)
![image](https://user-images.githubusercontent.com/16653571/157827768-f8f22d56-0e06-4051-ac9a-b4dd5a17b04a.png)
![image](https://user-images.githubusercontent.com/16653571/157827887-9dfa3804-426c-48be-afcb-fc422d1a5030.png)
#### After PR changes
![image](https://user-images.githubusercontent.com/16653571/157828148-b7f51ba6-5b29-476a-972f-4bce70273d0a.png)
![image](https://user-images.githubusercontent.com/16653571/157828569-b04a94b8-1c11-408c-a9d8-15180ff88236.png)
![image](https://user-images.githubusercontent.com/16653571/157828430-d11c2a70-8189-4920-9cc4-691913ec9658.png)


#### Note for reviewers:
The 2nd screenshot in "Earlier" and "After PR changes" are different i.e, "After PR changes" doesn't show a "green tick" around the username because of two reasons:
1. It (someway) indicates the username field value is correct which is not true.
2. This indicates a client-side validation earlier rendered using djng-angular now we will have to write separate indicators for password and username.

I'm fine with adding such a (green tick) indicator for the username field if needed, open to suggestions!

DubeySandeep commented on marketing/templates/marketing/conversion.html:
```diff
@@ -20,7 +20,7 @@ <h1 class="page-title__text">Conversion data</h1>
-        <ul class="djng-form-errors">
+        <ul class="login-form-errors">
           {% for error in form.non_field_errors %}
           <li class="invalid">{{ error|last }}</li>
```
I'm testing this page as I don't think the `last` filter is needed here. (not related to changes in this PR)

DubeySandeep commented on registration/forms.py:
```diff
@@ -86,14 +86,15 @@ class Textarea(InputStyleMixin, forms.widgets.Textarea):
         label='Your email or username',
         help_text='You can enter either your username or your email to login.',
         widget=TextInput,
+        max_length=150
```
This looks like a magic number but I have just tried to add the client-side validation added by djng earlier, should we remove it?

mtyaka replied:
I'm guessing this comes from django contrib's auth `User` models, so I think we should leave it in.

DubeySandeep commented on requirements/base.in:
```diff
@@ -12,8 +12,6 @@ cookies<2.3
 dictdiffer<0.9
 Django<2.3,>=2.2
-# using the latest django-angular (2.2 or later) requires a heavy rewrite of the registration form, so we use the old 1.0.0 version patched to work under Django 2
--e git+https://github.com/open-craft/django-angular.git@v1-with-django2-support#egg=django_angular
```
Working on updating the `base.txt` and `dev.txt` files. Facing some issue with `pip-compile` command.

DubeySandeep replied:
Done!

DubeySandeep replied:
Done!

DubeySandeep commented on marketing/templates/marketing/conversion.html:
```diff
@@ -20,9 +20,9 @@ <h1 class="page-title__text">Conversion data</h1>
       {% if form.non_field_errors %}
       <div class="page-content__inner__wrapper">
-        <ul class="djng-form-errors">
+        <ul class="login-form-errors">
```
`login-form-errors` looks unexpected here but I think it's added here to avoid duplicate CSS code. A possible solution to fix it (if needed):
- Change the class name to be more generic (custom-form-errors OR ocim-form-error)

mtyaka replied:
I think changing this to a more generic name makes sense 👍

DubeySandeep replied:
Done! (Change it to `ocim-form-errors`)

mtyaka commented on registration/templates/registration/form_field.html:
```diff
@@ -1,14 +1,22 @@
+                            {% for error in field.errors %}
+                                <li class="invalid">{{ error }}</li>
+                            {% endfor %}
+                        {% endif %}
```
Indentation looks a bit off here.

DubeySandeep replied:
Done!

mtyaka reviewed (changes requested):
I added a couple of small comments, and I didn't test this yet (will do it on Ocim stage once it's functional again), but otherwise this is looking really good @DubeySandeep :)

mtyaka commented:
The frontend-coverage tests failed, but it could be just a flaky test. I triggered a rerun.

mtyaka commented:
Yup, it was just a flaky test; not related to the changes in this PR.

DubeySandeep commented:
@mtyaka I've addressed your review comment, PTAL!

Also, please check the `Note for reviewers:` section (At the bottom of the PR description) while testing the changes. Do let me know if you have any suggestions/questions.

mtyaka commented:
The changes look good @DubeySandeep! I didn't test the code yet, because Ocim stage is currently still in a broken state, but will do it as soon as it's up and running again.

>  Also, please check the Note for reviewers: section (At the bottom of the PR description) while testing the changes. Do let me know if you have any suggestions/questions.

I agree with you that the green checkmark doesn't really make much sense for the username. No need to bring it back.

DubeySandeep commented:
> I didn't test the code yet, because Ocim stage is currently still in a broken state, but will do it as soon as it's up and running again.

Sure, sg! Just in case the stage server takes more time, I think (not sure!) we can test these changes locally as the changes are mostly in the backend/manage server, so running `make run.dev` and accessing localhost:5000/login will load the login form.

mtyaka commented:
This is good to merge, nice work @DubeySandeep :+1:

- [x] I tested this: I tested the login form on Ocim stage and verified that it works correctly.
- [x] I read through the code
- [x] I checked for accessibility issues
